org.eclipse.xtext.ui.tests/src-longrunning/org/eclipse/xtext/ui/tests/editor/DirtyStateEditorSupportIntegrationTest.java

```diff
@@ -122,18 +124,28 @@ private IEObjectDescription getFirstExportedObjectInLastEventDelta() {
 	}
 
 	protected void pushKey(char c, int k) throws Exception {
+		StyledText textWidget = editor.getInternalSourceViewer().getTextWidget();
+		
 		String textBefore = editor.getDocument().get();
 		Event event = new Event();
 		event.type = SWT.KeyDown;
 		event.character = c;
 		event.keyCode = k;
-		myDisplay.post(event);
+		event.doit = true;
+		if (c == SWT.ESC) {
+			event.keyCode = 27;
+		}
+		textWidget.notifyListeners(SWT.KeyDown, event);
```

Comment:
am not sure if this keeps the intention of the test

Reply:
I think it does.